src/main/java/baritone/behavior/MemoryBehavior.java

```diff
@@ -166,7 +174,26 @@ public void onBlockInteract(BlockInteractEvent event) {
 
     @Override
     public void onPlayerDeath() {
+        Waypoint deathWaypoint = new Waypoint("death", Waypoint.Tag.DEATH, ctx.playerFeet());
         baritone.getWorldProvider().getCurrentWorld().getWaypoints().addWaypoint(new Waypoint("death", Waypoint.Tag.DEATH, ctx.playerFeet()));
```

Comment:
change this to `baritone.getWorldProvider().getCurrentWorld().getWaypoints().addWaypoint(deathWaypoint);`